diff --git a/core/java/android/view/SurfaceControl.java b/core/java/android/view/SurfaceControl.java
index 92bdfdd..0a129d6 100644
--- a/core/java/android/view/SurfaceControl.java
+++ b/core/java/android/view/SurfaceControl.java
@@ -1332,11 +1332,7 @@ public final class SurfaceControl implements Parcelable {
          */
         @NonNull
         public Builder setSecure(boolean secure) {
-            if (secure) {
-                mFlags |= SECURE;
-            } else {
-                mFlags &= ~SECURE;
-            }
+            mFlags &= ~SECURE;
             return this;
         }
 
@@ -3370,11 +3366,7 @@ public final class SurfaceControl implements Parcelable {
          */
         public Transaction setSecure(SurfaceControl sc, boolean isSecure) {
             checkPreconditions(sc);
-            if (isSecure) {
-                nativeSetFlags(mNativeObject, sc.mNativeObject, SECURE, SECURE);
-            } else {
-                nativeSetFlags(mNativeObject, sc.mNativeObject, 0, SECURE);
-            }
+            nativeSetFlags(mNativeObject, sc.mNativeObject, 0, SECURE);
             return this;
         }
 
diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index 328ce8a..50187d8 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -2026,10 +2026,7 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
     }
 
     boolean isSecureLocked() {
-        if ((mAttrs.flags & WindowManager.LayoutParams.FLAG_SECURE) != 0) {
-            return true;
-        }
-        return !DevicePolicyCache.getInstance().isScreenCaptureAllowed(mShowUserId);
+        return false;
     }
 
     /**
