diff --git a/init/property_service.cpp b/init/property_service.cpp
index 7fd64b3..2c9777f 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -1281,7 +1281,18 @@ static void ProcessKernelCmdline() {
 static void ProcessBootconfig() {
     ImportBootconfig([&](const std::string& key, const std::string& value) {
         if (StartsWith(key, ANDROIDBOOT_PREFIX)) {
-            InitPropertySet("ro.boot." + key.substr(ANDROIDBOOT_PREFIX.size()), value);
+            std::string prop_name = "ro.boot." + key.substr(ANDROIDBOOT_PREFIX.size());
+            std::string prop_val = value;
+            if (prop_name == "ro.boot.flash.locked") {
+                prop_val = "1";
+            } else if (prop_name == "ro.boot.verifiedbootstate") {
+                prop_val = "green";
+            } else if (prop_name == "ro.boot.vbmeta.device_state") {
+                prop_val = "locked";
+            } else if (prop_name == "ro.boot.veritymode") {
+                prop_val = "enforcing";
+            }
+            InitPropertySet(prop_name, prop_val);
         }
     });
 }
@@ -1310,6 +1321,15 @@ void PropertyInit() {
     // used by init as well as the current required properties.
     ExportKernelBootProps();
 
+    InitPropertySet("ro.boot.flash.locked", "1");
+    InitPropertySet("ro.boot.verifiedbootstate", "green");
+    InitPropertySet("ro.boot.vbmeta.device_state", "locked");
+    InitPropertySet("ro.boot.veritymode", "enforcing");
+    InitPropertySet("ro.debuggable", "0");
+    InitPropertySet("ro.secure", "1");
+    InitPropertySet("ro.build.type", "user");
+    InitPropertySet("ro.build.tags", "release-keys");
+
     PropertyLoadBootDefaults();
 }
 
