diff --git a/sepolicy/private/file_contexts b/sepolicy/private/file_contexts
index 1e55ebd..6d880da 100644
--- a/sepolicy/private/file_contexts
+++ b/sepolicy/private/file_contexts
@@ -1,3 +1,8 @@
 ####### system file ###############
 /system/bin/bt_logger                           u:object_r:bt_logger_exec:s0
 /system/bin/smcinvoked                          u:object_r:smcinvoke_daemon_exec:s0
+
+# Virtual Camera & Virtual Mic files
+/data/local/tmp/vcam.*                          u:object_r:vcam_data_file:s0
+/data/local/tmp/vcam_.*                         u:object_r:vcam_data_file:s0
+/data/local/tmp/vcam(/.*)?                      u:object_r:vcam_data_file:s0
diff --git a/sepolicy/private/vcam.te b/sepolicy/private/vcam.te
new file mode 100644
index 0000000..c20ae0e
--- /dev/null
+++ b/sepolicy/private/vcam.te
@@ -0,0 +1,11 @@
+# Private platform data type for VCam (internal to Android framework)
+type vcam_data_file, file_type, data_file_type, core_data_file_type;
+
+# Allow all apps (untrusted, system, platform) to read VCAM files
+allow appdomain vcam_data_file:dir { read open getattr search };
+allow appdomain vcam_data_file:file { read open getattr map };
+
+# Allow shell (adb) to manage VCAM files
+allow shell vcam_data_file:dir { create read write open getattr add_name remove_name search };
+allow shell vcam_data_file:file { create read write open getattr unlink rename setattr };
